databricks_cli/dbfs/api.py

```diff
@@ -127,3 +127,100 @@ def mkdirs(self, dbfs_path):
 
     def move(self, dbfs_src, dbfs_dst):
         self.client.move(dbfs_src.absolute_path, dbfs_dst.absolute_path)
+
+    def copy_to_dbfs_non_recursive(self, src, dbfs_path_dst, overwrite):
+        # Munge dst path in case dbfs_path_dst is a dir
+        try:
+            if self.get_status(dbfs_path_dst).is_dir:
+                dbfs_path_dst = dbfs_path_dst.join(os.path.basename(src))
+        except HTTPError as e:
+            if e.response.json()['error_code'] == DbfsErrorCodes.RESOURCE_DOES_NOT_EXIST:
+                pass
+            else:
+                raise e
+        self.put_file(src, dbfs_path_dst, overwrite)
+
+    def copy_from_dbfs_non_recursive(self, dbfs_path_src, dst, overwrite):
+        # Munge dst path in case dst is a dir
+        if os.path.isdir(dst):
+            dst = os.path.join(dst, dbfs_path_src.basename)
+        self.get_file(dbfs_path_src, dst, overwrite)
+
+    def copy_to_dbfs_recursive(self, src, dbfs_path_dst, overwrite):
```

Comment:
nit: can we make these private? `copy_to_dbfs_recursive` -> `_copy_to_dbfs_recursive`

Reply:
yes